Source/FieldSolver/SpectralSolver/SpectralAlgorithms/SpectralBaseAlgorithm.H

```diff
@@ -11,6 +11,7 @@
 #include "Utils/WarpX_Complex.H"
 
 #include "FieldSolver/SpectralSolver/SpectralFieldData_fwd.H"
+#include "FieldSolver/SpectralSolver/SpectralFieldData.H"
```

Comment:
I thought the forward declaration header could be enough, but it does not seem to compile successfully without the full header added here (I guess because the member `m_spectral_index` added below is not simply a pointer to an instance of the new class `SpectralFieldIndex`).